controllers/convoController.js

```diff
@@ -104,10 +102,64 @@ const fetchAllMyConversationsGet = async (req, res) => {
 
 // Get a specific conversation and its messages
 const fetchConversationGet = async (req, res) => {
+	// Pulling out our conversation id and whether to translate or not
+	const { id, translate } = req.query;
+
 	try {
 		// Finding all messages associated with a specific conversation
-		const conversation = await Message.find({ conversationID: req.params.id });
-		res.send(conversation);
+		let messages = await Message.find({ conversationID: id });
+
+		if (translate === "true") {
+			// Contains recipient's messages and recipient's messages that need to be translated
+			let recipientMessages = [],
+				filteredRecipientMessages = [];
+
+			// Storing the recipient's messages to be translated
+			for (key in messages) {
+				if (String(req.user._id) !== String(messages[key].sender)) {
+					recipientMessages = [...recipientMessages, messages[key]];
+				}
+			}
+
+			// Filters in messages that need to be translated
+			recipientMessages.forEach((message) => {
+				if (!message.languages.includes(req.user.language)) {
```

Comment:
you could also move this logic down so for example in worst case scenario, everything is not translated so it would be O(N), N being the total num of `recipientMessages`, and then in `filteredRecipientMessages`, you'll have to do another loop to actually translate it

you can just loop through `recipientMessages`, and combine it with your translation and check if it needs to be translated it from line 142 to 145

Reply:
Thanks Bonnie, I'll implement that in a future PR!